superset/result_set.py

```diff
@@ -137,6 +138,9 @@ def __init__(  # pylint: disable=too-many-locals
         if array.size > 0:
             for column in column_names:
                 try:
+                    # Check if array[column][0] is an list of instance of psycopg2.extras.DateTimeTZRange
+                    if isinstance(array[column][0], list) and isinstance(array[column][0][0], DateTimeTZRange):
+                        array[column] = [[(y.lower, y.upper)  for y in x] for x in array[column]]
```

Comment:
Engine specific code shouldn't live in this file. Instead this should be a generic method that lives in db_engine_specs interface.

Reply:
> Instead this should be a generic method that lives in db_engine_specs interface.

Ah, did get you? I thought it should be in Postgres spec.
Should I write a generic method which can convert DateTimeTZRange (which is a specific data type comes with postgres adopter) to more generic as tuple of DataTime?

Reply:
in postgrestDBSpec

Reply:
@nytai I have tried this, but it's not working. Detailed explanation provided.